packages/marshal/src/helpers/copyRecord.js

```diff
@@ -24,25 +24,32 @@ export const CopyRecordHelper = harden({
   styleName: 'copyRecord',
 
   canBeValid: (candidate, check) => {
-    const reject = details => check(false, details);
     const proto = getPrototypeOf(candidate);
     if (proto !== objectPrototype && proto !== null) {
-      return reject(X`Unexpected prototype for: ${candidate}`);
+      return !!check && check(false, X`Unexpected prototype for: ${candidate}`);
```

Comment:
Interesting that this causes you to revert the `reject` refactoring.

No change suggested.

Reply:
I thought so too. There's a lesson here regarding tagged templates which I cannot fully articulate, although it's also interesting to note that we could alternatively have something like
```js
const reject = (strings, ...args) => !!check && check(false, X(strings, ...args))
…
if (!ok) {
  return reject`Unexpected prototype for: ${candidate}`;
}
```
for the loss of call-site-local `X` and the (hopefully minor) cost of array construction/spread.